hs_core/hydroshare/utils.py

```diff
@@ -303,6 +304,13 @@ def replace_resource_file_on_irods(new_file, original_resource_file):
             destination_file = os.path.join(ori_res.resource_federation_path, ori_res.short_id,
                                             original_resource_file.fed_resource_file_name_or_path)
     istorage.saveFile(new_file, destination_file, True)
+    if user is not None and user.is_authenticated():
+        last_changed_by = user
+    else:
+        last_changed_by = ori_res.last_changed_by
```

Comment:
It is probably not needed anymore. At one point we had the requirment to update the sqlite file prioir to bag or sqlite file download if the system detects that the sqlite file is out of sync. In that case it was a possiblity for the ```else`` part to happen if an authorised user tries to download the sqlite file or the bag for a public time series resource.